### PR TITLE
Add peek cards superpower

### PR DESCRIPTION
**Pull request by Dammic**
This PR adds 'peek 3 cards' superpower, which is available on small board.
The president will see 3 cards that the next president will draw
acknowl
next

### CODE REVIEW

File: events/SocketEvents.js [hunk 1]
```diff
@@ -375,6 +378,7 @@ module.exports = function (io) {
         const clientVerificationHof = ClientVerificationHof(RoomsManager)
         phaseSocketEvents.startGame = clientVerificationHof(['isOwner'], phaseSocketEvents.startGame)
         socketEvents.kickPlayer = clientVerificationHof(['isOwner'], socketEvents.kickPlayer)
+        socketEvents.kickPlayer = clientVerificationHof(['isPresident'], socketEvents.endPeekCardsPhase)
```
Comment (Terverok) [inline events/SocketEvents.js]: `kickPlayer` wtf? typo?:smiley:

File: utils/RoomsManager.js [hunk 2]
```diff
@@ -487,18 +487,24 @@ class RoomsManager {
...
-        let tmpDrawPile = drawPile
-        if (size(drawPile) < amount) {
-            tmpDrawPile = shuffle(concat(drawPile, discardPile))
+        const policies = take(drawPile, amount)
+        this.rooms_props[roomName].drawPile = drop(drawPile, amount)
+        this.rooms_props[roomName].drawnCards = policies
+
+        const drawPileLength = size(this.rooms_props[roomName].drawPile)
+        if (drawPileLength < amount || drawPileLength < 3) {
+            this.rooms_props[roomName].drawPile = shuffle(concat(this.rooms_props[roomName].drawPile, discardPile))
```
Comment (Terverok) [inline utils/RoomsManager.js]: In this case wouldn't be better to assign room to some variable eg. `const room = this.room_props[roomName]`, you don't use destructured variables often.

File: src/utils/SocketHandler.js [hunk 3]
```diff
@@ -267,6 +266,22 @@ export class SocketHandler extends React.PureComponent {
         socket.on(SocketEvents.SetTimer, ({ data: { waitTime } }) => {
             this.props.roomActions.setWaitTime({ waitTime })
         })
+        socket.on(SocketEvents.StartPeekCardsPhase, ({ data: { timestamp } }) => {
```
Comment (Terverok) [inline src/utils/SocketHandler.js]: If we doesn't change anything else maybe we could just send event that dispalys message?
That way we woudn't duplicate unnecessary events.

File: utils/__tests__/RoomsManager.spec.js [hunk 4]
```diff
@@ -447,4 +447,70 @@ describe('RoomsManager', () => {
...
+            initialRoomProps.drawnCards = [PolicyCards.FacistPolicy]
+            RoomsManager.takeChoicePolicyCards('testRoom', 1)
+
+            expect(RoomsManager.rooms_props['testRoom']).toEqual(initialRoomProps)
+        })
+        test('Should take 2 out of 4 cards and shuffle the rest with discards', () => {
+            RoomsManager.rooms_props['testRoom'].drawPile = [PolicyCards.FacistPolicy, PolicyCards.FacistPolicy, PolicyCards.LiberalPolicy, PolicyCards.LiberalPolicy]
+            RoomsManager.rooms_props['testRoom'].discardPile = [PolicyCards.FacistPolicy, PolicyCards.LiberalPolicy, PolicyCards.LiberalPolicy]
+            const initialRoomProps = cloneDeep(RoomsManager.rooms_props['testRoom'])
+            initialRoomProps.drawPile = RoomsManager.rooms_props['testRoom'].drawPile
```
Comment (Terverok) [inline utils/__tests__/RoomsManager.spec.js]: This line is overwritten later, you can remove it

File: utils/__tests__/RoomsManager.spec.js [hunk 5]
```diff
@@ -447,4 +447,70 @@ describe('RoomsManager', () => {
...
+            initialRoomProps.discardPile = []
+            initialRoomProps.drawnCards = [PolicyCards.FacistPolicy, PolicyCards.FacistPolicy]
+            RoomsManager.takeChoicePolicyCards('testRoom', 2)
+
+            initialRoomProps.drawPile = RoomsManager.rooms_props['testRoom'].drawPile
```
Comment (Terverok) [inline utils/__tests__/RoomsManager.spec.js]: This doesn't look too good :neutral_face: Maybe before that assignment we should check if `drawPile` array includes some expected cards.

File: utils/__tests__/RoomsManager.spec.js [hunk 6]
```diff
@@ -447,4 +447,70 @@ describe('RoomsManager', () => {
...
+            expect(RoomsManager.rooms_props['testRoom']).toEqual(initialRoomProps)
+            expect(RoomsManager.rooms_props['testRoom'].drawPile.length).toEqual(5)
+        })
+
+        test('Should take 1 out of 1 cards and shuffle the rest with discards', () => {
+            RoomsManager.rooms_props['testRoom'].drawPile = [PolicyCards.LiberalPolicy]
+            RoomsManager.rooms_props['testRoom'].discardPile = [PolicyCards.FacistPolicy, PolicyCards.LiberalPolicy, PolicyCards.LiberalPolicy]
+            const initialRoomProps = cloneDeep(RoomsManager.rooms_props['testRoom'])
+            initialRoomProps.drawPile = RoomsManager.rooms_props['testRoom'].drawPile
```
Comment (Terverok) [inline utils/__tests__/RoomsManager.spec.js]: same as above

File: utils/__tests__/RoomsManager.spec.js [hunk 7]
```diff
@@ -447,4 +447,70 @@ describe('RoomsManager', () => {
...
+            initialRoomProps.discardPile = []
+            initialRoomProps.drawnCards = [PolicyCards.LiberalPolicy]
+            RoomsManager.takeChoicePolicyCards('testRoom', 1)
+
+            initialRoomProps.drawPile = RoomsManager.rooms_props['testRoom'].drawPile
```
Comment (Terverok) [inline utils/__tests__/RoomsManager.spec.js]: same as in previous test case

File: utils/__tests__/RoomsManager.spec.js [hunk 8]
```diff
@@ -447,4 +447,70 @@ describe('RoomsManager', () => {
...
+        test('Should take 1 out of 3 cards and shuffle the rest with discards', () => {
+            RoomsManager.rooms_props['testRoom'].drawPile = [PolicyCards.FacistPolicy, PolicyCards.FacistPolicy, PolicyCards.LiberalPolicy]
+            RoomsManager.rooms_props['testRoom'].discardPile = [PolicyCards.FacistPolicy, PolicyCards.LiberalPolicy, PolicyCards.LiberalPolicy]
+            const initialRoomProps = cloneDeep(RoomsManager.rooms_props['testRoom'])
+            initialRoomProps.drawPile = RoomsManager.rooms_props['testRoom'].drawPile
+            initialRoomProps.discardPile = []
+            initialRoomProps.drawnCards = [PolicyCards.FacistPolicy]
+            RoomsManager.takeChoicePolicyCards('testRoom', 1)
+
+            initialRoomProps.drawPile = RoomsManager.rooms_props['testRoom'].drawPile
```
Comment (Terverok) [inline utils/__tests__/RoomsManager.spec.js]: the same as above

File: utils/__tests__/RoomsManager.spec.js [hunk 9]
```diff
@@ -447,4 +447,70 @@ describe('RoomsManager', () => {
...
+            expect(RoomsManager.rooms_props['testRoom']).toEqual(initialRoomProps)
+            expect(RoomsManager.rooms_props['testRoom'].drawPile.length).toEqual(5)
+        })
+    })
+
+    describe('peekPolicyCards', () => {
+        test('Should return 3 cards but not modify anything', () => {
+            RoomsManager.rooms_props['testRoom'].drawPile = [PolicyCards.FacistPolicy, PolicyCards.FacistPolicy, PolicyCards.LiberalPolicy, PolicyCards.LiberalPolicy]
+            const initialRoomProps = cloneDeep(RoomsManager.rooms_props['testRoom'])
+            const peekedCards = RoomsManager.peekPolicyCards('testRoom', 1)
```
Comment (Terverok) [inline utils/__tests__/RoomsManager.spec.js]: "1" is unnecessary argument

File: events/SocketEvents.js (hunk 1 above)
Comment (Dammic) [reply events/SocketEvents.js]: Woo, that's a huge typo right here, it's fortunate you managed to spot that.
Changing it right away, thanks!

File: src/utils/SocketHandler.js (hunk 3 above)
Comment (Dammic) [reply src/utils/SocketHandler.js]: What this event should do is change selecting player. We already have function for sending a message, perhaps we should use it here.
As well as perhaps create a separate event just for setting selecting player (we have a few of superpower using that)

**Pull request merged**